### PR TITLE
deskflow: init at 1.17.1

### LINKED ISSUE #346459: Package request: deskflow
**Project description**

Deskflow is the pure GPL-2 OSS project that is now the upstream for the proprietary Synergy client.

**Metadata**

* homepage URL: [deskflow.org](https://deskflow.org/)
* source URL: https://github.com/deskflow/deskflow
* license: gpl2
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghost**
- Init deskflow at 1.17.1, Deskflow lets you share one mouse and keyboard between multiple computers on Windows, macOS and Linux, https://deskflow.org
- Fixes #346459
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/de/deskflow/package.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+stdenv.mkDerivation {
+  pname = "deskflow";
+
+  version = "1.17.0";
+
+  src = fetchFromGitHub {
+    owner = "deskflow";
+    repo = "deskflow";
+    rev = "44bd69fdc8df726909f35b7efeeebc554bfde1c5";
```
Comment (sikmir) [inline pkgs/by-name/de/deskflow/package.nix:35]:
```suggestion
    rev = "${version}+r1";
```

File: pkgs/by-name/de/deskflow/package.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    ninja
+    pkg-config
+    qt6.wrapQtAppsHook
+  ];
+
+  patchPhase = ''
+    runHook prePatch
+    sed -i '/set(CMAKE_INSTALL_PREFIX \/usr)/d' cmake/Packaging.cmake
+    runHook postPatch
+  '';
```
Comment (sikmir) [inline pkgs/by-name/de/deskflow/package.nix:50]:
```suggestion
  postPatch = ''
    substituteInPlace cmake/Packaging.cmake --replace-warn "set(CMAKE_INSTALL_PREFIX /usr)" ""
  '';
```

**Comment by char** [comment]
If anyone wants to run this as-is, it looks like deskflow wants libportal >= 0.8 and libei >= 1.3 for wayland so you'll need to provide those instead of the current versions, but libei is up to date on unstable and there's an input-leap related PR that will update libportal

**Review by jmir1 — APPROVED** [review]
Tested and works well on GNOME 46

File: pkgs/by-name/de/deskflow/package.nix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  postInstall = ''
+    substituteInPlace $out/share/applications/deskflow.desktop \
+        --replace-fail "Path=/usr/bin" "Path=$out/bin" \
+        --replace-fail "Exec=/usr/bin/deskflow" "Exec=deskflow"
+  '';
+
+  meta = {
+    homepage = "https://github.com/deskflow/deskflow";
+    description = "Deskflow lets you share one mouse and keyboard between multiple computers on Windows, macOS and Linux";
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deskflow/package.nix:82]:
```suggestion
    description = "Share one mouse and keyboard between multiple computers on Windows, macOS and Linux";
```

File: pkgs/by-name/de/deskflow/package.nix [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    mainProgram = "deskflow";
+    maintainers = with lib.maintainers; [ aucub ];
+    license = lib.licenses.gpl2Plus;
+    platforms = [
+      "x86_64-linux"
+      "aarch64-darwin"
+    ];
+    broken = stdenv.hostPlatform.isDarwin;
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deskflow/package.nix:90]:
```suggestion
    platforms = lib.platforms.linux;
```

File: pkgs/by-name/de/deskflow/package.nix [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    qt6.wrapQtAppsHook
+  ];
+
+  postPatch = ''
+    substituteInPlace cmake/Packaging.cmake --replace-warn "set(CMAKE_INSTALL_PREFIX /usr)" ""
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deskflow/package.nix:47]:
```suggestion
    substituteInPlace cmake/Packaging.cmake --replace-fail "set(CMAKE_INSTALL_PREFIX /usr)" ""
```

File: pkgs/by-name/de/deskflow/package.nix [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "deskflow";
+
+  version = "1.17.0";
+
+  src = fetchFromGitHub {
+    owner = "deskflow";
+    repo = "deskflow";
+    rev = "${version}+r1";
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deskflow/package.nix:35]: Where is this tag? There's only `v1.17.0`

File: pkgs/by-name/de/deskflow/package.nix [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  pugixml,
+  python3,
+  gdk-pixbuf,
+  libnotify,
+  qt6,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deskflow";
+
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deskflow/package.nix:29]:
```suggestion
```

File: pkgs/by-name/de/deskflow/package.nix (hunk 6 above)
Comment (aucub) [reply pkgs/by-name/de/deskflow/package.nix:35]: The old tag was deleted upstream

Comment (Aleksanaa) [reply pkgs/by-name/de/deskflow/package.nix:35]: This is not a good sign. You'd better write a commit has instead.

File: pkgs/by-name/de/deskflow/package.nix [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        --replace-fail "Exec=/usr/bin/deskflow" "Exec=deskflow"
+  '';
+
+  meta = {
+    homepage = "https://github.com/deskflow/deskflow";
+    description = "Share one mouse and keyboard between multiple computers on Windows, macOS and Linux";
+    mainProgram = "deskflow";
+    maintainers = with lib.maintainers; [ aucub ];
+    license = lib.licenses.gpl2Plus;
+    platforms = lib.platforms.linux;
```
Comment (Atemu) [inline pkgs/by-name/de/deskflow/package.nix:81]: Upstream docs say that it's supported on pretty much any platform.

```suggestion
```

If our build is broken on some of those, you should instead set `broken` or `badPlatforms` (there currently isn't an agreed upon standard here, I prefer `broken`).

File: pkgs/by-name/de/deskflow/package.nix [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    qt6.wrapQtAppsHook
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_SKIP_RPATH=ON"
```
Comment (Atemu) [inline pkgs/by-name/de/deskflow/package.nix:46]: Why is this necessary? Please add a comment.

File: pkgs/by-name/de/deskflow/package.nix [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    libXi
+    libXrandr
+    libXtst
+    libxkbcommon
+    pugixml
+    gdk-pixbuf
+    libnotify
+    python3
+    qt6.qtbase
+  ];
```
Comment (Atemu) [inline pkgs/by-name/de/deskflow/package.nix:67]: Since you've separated the deps so nicely, you should set `strictDeps = true;`.

**Review by Atemu — APPROVED** [review]
Diff generally LGTM and if it works on GNOME, I'd say that's quite the achievement already.

A couple of small things, then we're good to go I think.

File: pkgs/by-name/de/deskflow/package.nix [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    repo = "deskflow";
+    rev = "v${version}";
+    hash = "sha256-cEKG9MwENbZqrfRdwiZtRWmIfRndrWUoaZQ5O7YRpBs=";
+  };
+
+  postPatch = ''
+    substituteInPlace src/lib/deskflow/unix/AppUtilUnix.cpp \
+      --replace-fail "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboard_config}/share/X11/xkb/rules/evdev.xml"
+    substituteInPlace src/lib/gui/tls/TlsCertificate.cpp \
+      --replace-fail "\"openssl\"" "\"${openssl.bin}/bin/openssl\""
```
Comment (sikmir) [inline pkgs/by-name/de/deskflow/package.nix:48]:
```suggestion
      --replace-fail "\"openssl\"" "\"${lib.getBin openssl}/bin/openssl\""
```

**Comment by x-zvf** [comment]
Tested and working on Plasma6/Wayland

**Review by Aleksanaa — APPROVED** [review]
